bildb/lib/screens/locations.ml

```diff
@@ -180,27 +181,15 @@ module Make (Machine : Primus.Machine.S) = struct
         | Some value' ->
           begin
 
-              (* Is the address we're going to write to unmapped?
-                 If not, we can write to it. *)
-              let%bind addrs = unmapped addr' ~addr_size ~num_words:1 in
-              match addrs with
-              | [] ->
-                begin      
-                  let%bind _ = Mem.store addr' value' in
-                  show_locs addr ~prompt ~handler ~num_words:"1"
-                end
-
-              (* If the address isn't mapped, alert the user. 
-                 Trying to write to it would result in a page fault,
-                 and cause Primus to exit. *)
-              | _ ->
-                begin
-                  let msg = Printf.sprintf 
-                    "The address is unmapped and can't be written to: %s" 
-                    (Utils.string_of addr') in
-                  let text = [Ui.mk_output ~color:Tty.Red msg] in
-                  Machine.return (Event.screen () ~text ~prompt ~handler)
-                end
+              (* Allocate the memory at the specified address (if needed),
+                 then store the provided value there. *)
+              let* is_mapped = Mem.is_mapped addr' in
```

Comment:
Could you give a brief description of the code being added here?... and/or more generally your use of `Machine.return` and `Mem.allocate`? 
(By brief description, I just mean in this pr, not actually adding comments in the code.)

Reply:
In general, all things we do in Primus are monadic actions. They have "side-effects" (I change values in memory, or in a register, or whatever). So they're all monadic types. Hence, any value we want to return must be returned with `Machine.return (value)`, which is the monadic return that the `Machine` module provides for us. 

Most of the time, when I make some change to the memory or registers, I don't need to return a value, so I finish that action with `Machine.return ()`.

`Memory.allocate` already has built into it that after it has allocated the memory, it will call `Machine.return ()`, so I don't have to actually call that. It will do it for me. 

In this particular place, both branches of this if-statement return the same type. On one branch, we do nothing (i.e., we return `Machine.return ()`). On the other branch, we allocate a bit of memory, and then do nothing (i.e., let `Memory.allocate` return `Machine.return ()`).

Does that help any?

Reply:
That helps a great deal! Thanks.